lib/Dialect/Arith/Conversions/ArithToCGGI/ArithToCGGI.cpp

```diff
@@ -220,15 +295,20 @@ struct ArithToCGGI : public impl::ArithToCGGIBase<ArithToCGGI> {
         });
 
     target.addDynamicallyLegalOp<memref::AllocOp>([&](Operation *op) {
-      // Check if all Store ops are constants, if not store op, accepts
-      // Check if there is at least one Store op that is a constants
+      // Check if all Store ops are constants or GetGlobals, if not store op,
+      // accepts Check if there is at least one Store op that is a constants
       return (llvm::all_of(op->getUses(),
```

Comment:
nit: can we split this condition up into multiple lines to make it more readable?